### DISCUSSION TITLE
Hello from StackOverflow; looks great

### DISCUSSION THREAD

**Discussion by Baumflaum** [Escain/QTopMenu#2] [General]
Hi, seeing your example code I was also assuming you were a Qt beginner. But looking at your brainchild I'd say we are on about the same skill level! Given your StackOverflow example everything is said with "I attributed everything I don't like to the minimal-working-exaple-nature of your code."

**Comment by Escain** [comment]
Thank you for having a look :-)

**Comment by Baumflaum** [comment]
I'll edit it to be less condescending. Wasn't sure when typing (wasn't my intention) but I see how one could feel attacked. 

What are you writing this code for?

**Reply by Escain** [reply]
I did not feel attaqued at all. Don't worry :-)
Actually, I am still interested in you though about the "minimal code" that I posted on the question: I know its not release-quality, but I wonder what are your main concerns about it (so I can post better questions next time). If you don't mind to write them down in 2-3 lines.

> What are you writing this code for?

I am writting a CAM software for my milling machine, and the top menu is always a problem: Is not that trivial to implement as it could seems, and there are not that many implementations open-source out there.
So I focusses on some requirements:
- Cross-platform
- Open-source license (even if I keep a proprietary source-base too)
- Colors which adapts to the OS scheme, including icons
- A zone for buttons which are tab-independent  (e.g. you can "save" even if your are on a different tab)
- Possibility for the end-user to customize some areas (not yet done, but all is ready for it)
- Possibility to have custom widgets there (currently only the push-button is implemented).
- The menu can be placed Vertical (Left-side), or Horizontal (normal top side)
- And all the standard: tab focus management, hover/pressed/focus/disabled styles, shortcuts for tabs, different widget layouts

**Reply by Baumflaum** [reply]
The problem with the code was the unusual structure of the widgets. In most cases children of widgets aren't accessible outside of the widgets themselves. So "implanting" your t1-t4 from main and connecting the lambda there was weird. A better solution would have been to just write all the code you had in your main inside your TestWidget. 

That is a cool project. I'm also often struggling with the basicness of Qt Widgets. For simple "however it looks" applications it's great; but anything more specific and you are in a world of pain.
